README.md

```diff
@@ -17,94 +25,136 @@
- 
-### Chapter 2: Wrangling data
+- Using an existing pandas dataframe
+- From excel or csv to pandas df
+- Creating our own pandas df 
+- Using data from an API
+
+### Chapter 7: Wrangling data
 
- - Intro to Pandas's basic operations
- - wrangling an example dataset 
+- Case Study: Select a dataset
+- Basic operations with pandas
+- Cleaning, processing, wrangling
+- Preprocessing files
 
-### Chapter 3: Data Visualization
+### Chapter 8: Data Visualization
 
- - Intro to Plotly Express (PX)
- - PX graphs: pie chart, bar chart, scatter plot (using fig.show())
- - Incorporting PX graphs in Dash app 
+ - Introduction to Plotly Express
+ - Plotly Express graphs
+   - scatter plot
+   - line plot
+   - bar plot
+   - etc...
+ - Principles of Effective Visualizations
+ - Incorporting PX graphs in a Dash app 
+ - References and resourcres
 
-## Part III: Dash
+## Part III: Polishing your Dash application
 
- - What you will learn in this part
+- Summary and overview of this part
+- Learning Intentions
  
-### Chapter 1: Advanced Layout and Styling
+### Chapter 9: Advanced Layout and Styling
 
  - Creating a layout component inside a callback
- - Dash Bootstrap Components: dbc.Container, dbc.Row, dbc.Col
- - Dash Bootstrap Components: Accordion, alert, input
+ - More Dash Bootstrap Components
+ - Dash HTML components
+ - CSS (Optional)
   
-### Chapter 2: Advanced Callback
+### Chapter 10: Advanced Callbacks
 
- - State
- - multiple buttons: callback_context
+ - States
+ - Multiple buttons: callback_context
  - Multiple outputs and inputs
 
-### Chapter 3: Advanced Components
+### Chapter 11: Advanced Components
 
  - RadioItem
- - RagneSlider
+ - RangeSlider
  - Datepicker
  - Interval
  - Store
 
-### Chapter 4: DataTable
+### Chapter 12: DataTable
 
  - Intro to the DataTable: creating a basic DataTable
  - Linking dataTable to graph
  - Editing the DataTable
  - Other importnat DataTable props
 
-## Part IV: Important stuff
+## Part IV: Publishing your Dash Application
 
- - What you will learn in this part
+- What you will learn in this part
+- Learning Intentions
 
-### Chapter 1: Proper Deployment
+### Chapter 13: Full Deployment
 
- - How to properly deploy your app to heroku
- - Should we add subsection on deploying to Pythoanywhere??????
+ - Components of deployment
+   - Procfile
+   - requirements.txt
+   - Dockerfile
+   - Heroku build pipeline 
+ - Deploying to Pythoanywhere ? (Optional)
 
-### Chapter 2: Improving app performance and multi-page app
+### Chapter 14: Improving app performance and multi-page app
 
- - Callback graph to asses speed
+ - Callback graphs to asses speed
  - Actions to improve app speed
  - Efficiency
  - Building multi-page app
  
-### Chapter 3: Gallery
-
- - Need to add something here?????????
-
-### Chapter 4: Case Study
-
- - Need to add something here?????????
+### Chapter 15: Gallery and Template
+
+- [Dashboard Layouts](https://github.com/matthewconnell/dashr_sample_layouts)
```

Comment:
I see a very useful layouts section as well as case studies. Do we need to add a section on Intro to Gallery? Or what exactly do we mean by Gallery in the headline of Chapter 15?

Reply:
Do you mean you want to have separate chapters? One for Gallery, and one for templates?

Reply:
not necessarily. I don't think I want anything yet. I was just trying to understand. By Gallery, we mean that we will introduce the gallery of beginner apps to the reader, and encourage the reader to add their app to it?

Reply:
Wth the gallery we'll show some more curated "beginner" apps that people have created (To be discussed where these apps will come from, one option could be the "best" apps that students in our classes create, but could also be community apps as well).

With the templates, what I want is a barebones app setup that people can clone and edit with just "stubs" for where different components are etc...

Reply:
Got you. Thanks for clarifying.